### DISCUSSION TITLE
Support for PREEMPT_RT kernel

### DISCUSSION THREAD

**Discussion by tpimh** [johang/sd-card-images#77] [General]
I am using RK3399, and I want to run a realtime kernel. Rockchip seems to provide source for Linux 4.4.176, 4.19.87, 4.19.104 and 5.10.53 with PREEMPT_RT patches applied. Are there any precompiled real time kernels in the repositories?

**Comment by johang** [comment]
The builds are using official Debian/Ubuntu kernels. If you want patches or different config you'll have to build it yourself.

**Reply by tpimh** [reply]
Debian does provide [a config with PREEMPT_RT enabled](https://packages.debian.org/bookworm/linux-image-rt-arm64) in their source package.

**Reply by johang** [reply]
Interesting. Didn't know that. Try installing it and see if it's picked up correctly in `/boot/boot.cmd`. You may have to remove the regular kernel also (`apt remove linux-image-arm64`).
